bank-sdk/example-app/src/androidTest/java/net/gini/android/bank/sdk/exampleapp/ui/testcases/ExtractionScreenTests.kt

```diff
@@ -0,0 +1,70 @@
+package net.gini.android.bank.sdk.exampleapp.ui.testcases
+
+import android.Manifest
+import androidx.test.espresso.IdlingRegistry
+import androidx.test.ext.junit.rules.activityScenarioRule
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.GrantPermissionRule
+import androidx.test.uiautomator.UiDevice
+import net.gini.android.bank.sdk.exampleapp.ui.MainActivity
+import net.gini.android.bank.sdk.exampleapp.ui.resources.ImageUploader
+import net.gini.android.bank.sdk.exampleapp.ui.resources.SimpleIdlingResource
+import net.gini.android.bank.sdk.exampleapp.ui.screens.CaptureScreen
+import net.gini.android.bank.sdk.exampleapp.ui.screens.ExtractionScreen
+import net.gini.android.bank.sdk.exampleapp.ui.screens.MainScreen
+import net.gini.android.bank.sdk.exampleapp.ui.screens.OnboardingScreen
+import net.gini.android.bank.sdk.exampleapp.ui.screens.ReviewScreen
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+
+/**
+ * Test class for Extraction screen.
+ *
+ * Jira link for test case: [https://ginis.atlassian.net/browse/PM-25](https://ginis.atlassian.net/browse/PM-25)
+ */
+class ExtractionScreenTests {
+    @get:Rule
+    val activityRule = activityScenarioRule<MainActivity>()
+
+    @get: Rule
+    val grantPermissionRule: GrantPermissionRule =
+        GrantPermissionRule.grant(Manifest.permission.CAMERA)
+
+    private val mainScreen = MainScreen()
+    private val onboardingScreen = OnboardingScreen()
+    private val captureScreen = CaptureScreen()
+    private val imageUploader = ImageUploader()
+    private val reviewScreen = ReviewScreen()
+    private val extractionScreen = ExtractionScreen()
+    private lateinit var idlingResource: SimpleIdlingResource
+
+    private fun grantStoragePermission() {
+        val device: UiDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+        device.executeShellCommand("pm grant net.gini.android.bank.sdk.exampleapp android.permission.READ_EXTERNAL_STORAGE")
+        device.executeShellCommand("pm grant net.gini.android.bank.sdk.exampleapp android.permission.WRITE_EXTERNAL_STORAGE")
+    }
+
+    @Before
+    fun setup() {
+        grantStoragePermission()
+        idlingResource = SimpleIdlingResource(2000)
+        IdlingRegistry.getInstance().register(idlingResource)
+    }
+
+    @Test
+    fun test2_clickSendFeedbackAndCloseButton() {
```

Comment:
[Request] The title of this test is a bit ambiguous...it should be `test1`, and from the steps in the test it's a bit unclear which is the `sendFeedback` step, as there is no method with that name.

Also, could this test end in an assertion? Ideally the tests should end by checking something... I can only see the `assertReviewTitleIsDisplayed`, which does not seem to be related to the purpose of this test.

Reply:
This is suppose to be` test2_... ` because as per the test-case: https://ginis.atlassian.net/browse/PM-25, the test-step 1 is just the extraction of the data from the uploaded invoice on the screen...

Reply:
I've also wrote some questions to you regarding it already on slack, but i can copy/paste them here too:
- what do you mean by _there is no method_? are you talking about the source-code?
- shall i change the name of method to remove the ambiguity of what exactly the method is doing?

I can add more assertion in the end of method, but the test-result for this specific case (mentioned in PM-25, test-step 2) is similar to what I have implemented already